File: leetcode/3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> mp  =  new HashMap<>();
        int i = 0 , j = 0 , ans = 0;
        char[] c = s.toCharArray();
        while( j < c.length){
               if(mp.containsKey(c[j])){
                   ans = Integer.max(ans,j-i);
                   i = Integer.max(i,mp.get(c[j]) + 1);
                   
               }
               mp.put(c[j],j);
               j++;
        }
        ans = Integer.max(ans,j-i);
        return ans;
        
    }
}

File: leetcode/3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        unordered_map<char,int> mp;
        int left = 0,right = 0;
        int curr = 0,maxi = 0;
        while(right < s.length()){
            char c = s[right];
             mp[c]++;
             while(left <= right && mp[c] > 1){
                    mp[s[left]]--;
                    left++;
             }
             curr = right - left +1;
             maxi = max(curr,maxi);
             right++;

        }
        return maxi;
    }
};